src/example-types/metricGroups/pivot.js

```diff
@@ -52,6 +57,23 @@ let drilldownFilters = ({ drilldowns = [], groups = [], schema, getStats }) =>
     return drilldown && filter({ drilldown, ...group }, schema, getStats)
   }, groups)
 
+// Builds filters for drilldowns
+let paginationSkipFilters = ({
+  drilldowns = [],
+  skip = [],
+  groups = [],
+  schema,
+  getStats,
+}) => {
+  let group = groups[drilldowns.length]
+  if (!group || _.isEmpty(skip)) return false
+  let filter = lookupTypeProp(_.stubFalse, 'drilldown', group.type)
```

Comment:
All of this should happen inside the `_.map` because you can have different group types at each level. The bug with the current implementation would be paginating `fieldValues` at one level and a date range at another, which would cause both levels to be filtered using the `drilldown` method of the innermost grouping. You can solve this by simply moving these lines into the map and defining `group` based current index.

With all that said, I think you can simplify this even further and just reuse `drilldownFilters`. I'm pretty sure it does exactly what you need.

Reply:
All the skip filters are on the same level of grouping. We are mapping through the values we want to skip on the same level

Reply:
Just following up from our conversation - this is correct, the skip filter is operating on a series of values for the same group and is always operating on the last drilldown group 👍